### PR TITLE
Expose display cutout insets as well as the raw window insets

### LINKED ISSUE #681: Add support for display cutout insets
There seems to be no easy way to handle display cutout insets when using accompanist. It would be great if a `WindowInsets.displayCutout` property existed allowing you to handle cutouts too.

### PR DESCRIPTION
**Pull request by ansman**
This fixes #681

### CODE REVIEW

File: insets/src/main/java/com/google/accompanist/insets/WindowInsets.kt [hunk 1]
```diff
@@ -37,6 +40,13 @@ import androidx.core.view.WindowInsetsCompat
 @Stable
 interface WindowInsets {
 
+    /**
+     * The underlying [WindowInsetsCompat] that produced the insets.
+     *
+     * Prefer using one of the predefined types instead of this.
+     */
+    val rawInsets: WindowInsetsCompat
```
[the remark this answers is not in the document]
Comment (chrisbanes) [reply insets/src/main/java/com/google/accompanist/insets/WindowInsets.kt:48]: Not sure on this. We specifically want to keep Android types out of the API surface.

What's the use case?

**Review by chrisbanes — COMMENTED** [review]
Thanks for the PR

File: insets/src/main/java/com/google/accompanist/insets/WindowInsets.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (ansman) [reply insets/src/main/java/com/google/accompanist/insets/WindowInsets.kt:48]: I can agree with that. The use case would be more advanced display cutout handling such as knowing the exact path or something like that. For my use case the safe insets are typically fine.

Another option is to surface it as a separate composition local that is android specific. 

The problem in general is that a view can only have a single insets listener so if you use accompanist you cannot access the underlying insets and you have to wait for a PR to fix things that is missing from the API. 

I'm OK with removing it for now though.

**Comment by ansman** [comment]
I removed the raw insets.

File: insets/src/main/java/com/google/accompanist/insets/WindowInsets.kt [hunk 2]
```diff
@@ -369,6 +393,9 @@ private class InnerWindowInsetsAnimationCallback(
         if (animation.typeMask and WindowInsetsCompat.Type.systemGestures() != 0) {
             windowInsets.systemGestures.onAnimationStart()
         }
+        if (animation.typeMask and WindowInsetsCompat.Type.displayCutout() != 0) {
+            windowInsets.displayCutout.onAnimationStart()
```
[the remark this answers is not in the document]
Comment (chrisbanes) [reply insets/src/main/java/com/google/accompanist/insets/WindowInsets.kt:384]: Yep, Shyam

**Review by chrisbanes — APPROVED** [review]

**Comment by chrisbanes** [comment]
Thanks!

**Pull request merged**